### DISCUSSION TITLE
onFieldValidateEnd的执行时机问题

### DISCUSSION THREAD

**Discussion by li9269391** [alibaba/formily#1604] [Q&A 2.x] [answered]
https://codesandbox.io/s/quizzical-raman-fr91m?file=/App.tsx
字段设置了校验触发方式为 triggerType: 'onBlur' 失去焦点时校验，同时监听 effects onFieldValidateEnd 事件回调，但发现在 input change 的时候一直执行 onFieldValidateEnd 事件，并非预期的“监听某个字段校验触发结束的副作用钩子”。
包括其它 onFieldValidate 相关的勾子都没有关联 triggerType 来执行。
![image](https://user-images.githubusercontent.com/10287877/122628830-27e2d080-d0eb-11eb-98c9-a4bd136b2c4a.png)

**Comment by janryWang** [comment, marked as answer]
要分清，生命周期钩子和校验器的区别，你用生命周期钩子，它是遇到校验它就会执行的，且不区分triggerType，triggerType只是会控制校验器的执行时机，比如你自定义一个validator，同时指定triggerType，那么这个validator的执行时机就会受triggerType控制

**Reply by li9269391** [reply]
明白了，感谢

**Comment by li9269391** [comment]
![image](https://user-images.githubusercontent.com/10287877/122635083-735da480-d114-11eb-926f-ffedb5c70f9e.png)
分享一下：上面那个场景需求的解决方案
